pylint/checkers/magic_methods.py

```diff
@@ -0,0 +1,148 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from typing import TYPE_CHECKING
+
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class ManualMagicMethodChecker(BaseChecker):
+    """Check for manual __magic__ method calls."""
+
+    __implements__ = IAstroidChecker
+
+    # Docs: https://docs.python.org/3/reference/datamodel.html#basic-customization
+    # We exclude __init__, __new__, __subclasses__, __init_subclass__,
+    # __set_name__, __class_getitem__, __missing__, __exit__, __await__,
+    # __del__, __aexit__, __getnewargs_ex__, __getnewargs__, __getstate__,
+    # __setstate__, __reduce__, __reduce_ex__
+    # since these either have no alternative method of being called or
+    # have a genuine use case for being called manually.
```

Comment:
```suggestion
    """Check for manual __magic__ method calls.
    
    Docs: https://docs.python.org/3/reference/datamodel.html#basic-customization
    We exclude __init__, __new__, __subclasses__, __init_subclass__,
    __set_name__, __class_getitem__, __missing__, __exit__, __await__,
    __del__, __aexit__, __getnewargs_ex__, __getnewargs__, __getstate__,
    __setstate__, __reduce__, __reduce_ex__
    since these either have no alternative method of being called or
    have a genuine use case for being called manually.
    """

    __implements__ = IAstroidChecker
```